### ISSUE TITLE
Not seeing Fluentd records written to Elasticsearch

### ISSUE THREAD

**Issue by ewolinetz** [floragunncom/search-guard#158]
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)

sh-4.2$ cat /etc/system-release
CentOS Linux release 7.2.1511 (Core)

ES Version 2.3.3-1
SG Versions:
com.floragunn/search-guard-ssl/2.3.3.13
com.floragunn/search-guard-2/2.3.3.1

I am using ClientCert auth with NoOps.  My Fluentd roles are similar to the Logstash example:

sg_role_mapping:
```
sg_role_fluentd:
  users:
    - 'CN=system.logging.fluentd,OU=Logging,O=Logging,L=Test,C=DE'
```

sg_roles:
```
sg_role_fluentd:
  indices:
    '*':
      '*':
        - CRUD
        - CREATE_INDEX
```

sg_config:
```
searchguard:
  dynamic:
    http:
      xff:
        enabled: false
        internalProxies: 192\.168\.0\.10|192\.168\.0\.11
        remoteIpHeader: "x-forwarded-for"
        proxiesHeader: "x-forwarded-by"
        trustedProxies: "proxy1|proxy2"
    authc:
      authentication_domain_basic_internal:
        enabled: true
        order: 0
        authentication_backend:
          type: com.floragunn.searchguard.auth.internal.NoOpAuthenticationBackend
        authorization_backend:
          type: com.floragunn.searchguard.auth.internal.NoOpAuthorizationBackend
        http_authenticator:
          type: com.floragunn.searchguard.http.HTTPClientCertAuthenticator
          challenge: false
      authentication_domain_proxy:
        enabled: true
        order: 0
        authentication_backend:
          type: com.floragunn.searchguard.auth.internal.NoOpAuthenticationBackend
        authorization_backend:
          type: com.floragunn.searchguard.auth.internal.NoOpAuthorizationBackend
        http_authenticator:
          type: com.floragunn.searchguard.http.HTTPProxyAuthenticator
          challenge: false
          config:
            user_header: "X-Proxy-Remote-User"
            roles_header: "x-proxy-roles"
```

In my ES logs I see the following repeating itself over and over:
```
[2016-07-01 21:32:02,844][DEBUG][com.floragunn.searchguard.http.XFFResolver] xff resolve /172.17.0.3:48522
[2016-07-01 21:32:02,844][DEBUG][com.floragunn.searchguard.http.XFFResolver] no xff done true,false,class org.elasticsearch.http.netty.NettyHttpRequest,{}
[2016-07-01 21:32:02,844][DEBUG][com.floragunn.searchguard.auth.BackendRegistry] Try to extract auth creds from http clientcert 
[2016-07-01 21:32:02,844][DEBUG][com.floragunn.searchguard.auth.BackendRegistry] User 'CN=system.logging.fluentd,OU=Logging,O=Logging,L=Test,C=DE' is in cache? true (cache size: 1)
[2016-07-01 21:32:02,845][DEBUG][com.floragunn.searchguard.auth.BackendRegistry] User 'User [name=CN=system.logging.fluentd,OU=Logging,O=Logging,L=Test,C=DE, roles=[]]' is authenticated
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] evaluate permissions for User [name=CN=system.logging.fluentd,OU=Logging,O=Logging,L=Test,C=DE, roles=[]]
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] requested indices:data/write/bulk from 172.17.0.3:48522
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] Resolve [.operations-2016.07.01] from class org.elasticsearch.action.index.IndexRequest
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] indicesOptions IndicesOptions[id=48, ignore_unavailable=false, allow_no_indices=false, expand_wildcards_open=false, expand_wildcards_closed=false, allow_alisases_to_multiple_indices=false, forbid_closed_indices=true]
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] raw indices [.operations-2016.07.01]
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] Resolved [.operations-2016.07.01] to {}
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] Resolve [.operations-2016.07.01] from class org.elasticsearch.action.index.IndexRequest
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] indicesOptions IndicesOptions[id=48, ignore_unavailable=false, allow_no_indices=false, expand_wildcards_open=false, expand_wildcards_closed=false, allow_alisases_to_multiple_indices=false, forbid_closed_indices=true]
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] raw indices [.operations-2016.07.01]
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] Resolved [.operations-2016.07.01] to {}
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] Resolve [.operations-2016.07.01] from class org.elasticsearch.action.index.IndexRequest
[2016-07-01 21:32:02,860][DEBUG][com.floragunn.searchguard.configuration.PrivilegesEvaluator] indicesOptions IndicesOptions[id=48, ignore_unavailable=false, allow_no_indices=false, expand_wildcards_open=false, expand_wildcards_closed=false, allow_alisases_to_multiple_indices=false, forbid_closed_indices=true]
```

Am I missing something?  I've tried using the `ALL` action for my role mapping and confirmed that when not using Searchguard I see in my ES logs the records from Fluentd created.  I even reconfigured my Fluentd to send with the index 'operations.YYYY.MM.DD' to check that the preceeding '.' wasn't throwing off the role.

**Comment by floragunncom** [comment]
This was bug in 2.3.3.1 and is fixed with 2.3.3.2
See https://github.com/floragunncom/search-guard/issues/150

**Comment by ewolinetz** [comment]
I bumped up to 2.3.3.2 and I started seeing records created.  Thanks!
